### ISSUE TITLE
New-HPOVExternalRepository call results in disconnected session

### ISSUE THREAD

**Issue by daveolker** [HewlettPackard/POSH-HPOneView#357]
Please fill in as much information as possible to help resolve your issue.

### Expected Behavior
I am attempting to add an external repository using anonymous credentials as shown in Example 2 of the help text:

```PowerShell
    -------------------------- EXAMPLE 2 --------------------------
    New-HPOVExternalRepository -Name MyRepoName -Hostname webserver.domain.local -Directory OneViewRepo -Http
```    
    Configure an external repository using anonymous authentication, and HTTP protocol.

### Actual Behavior

When I attempt this on my system:

```PowerShell
PS > Connect-HPOVMgmt
cmdlet Connect-HPOVMgmt at command pipeline position 1
Supply values for the following parameters:
Hostname: 15.113.228.235
UserName: Administrator

ConnectionID Name           UserName      AuthLoginDomain Default
------------ ----           --------      --------------- -------
1            15.113.228.235 Administrator LOCAL           True   


PS > New-HPOVExternalRepository -Name "st052" -Hostname "gse-rose-st052.us.hpecorp.net" -Directory "/" -Http
New-HPOVExternalRepository : [Send-HPOVRequest]: **Your session has timed out or is not valid. Please use Connect-HPOVMgmt to authenticate to your appliance.**
At line:1 char:1
+ New-HPOVExternalRepository -Name "st052" -Hostname "gse-rose-st052.us ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : AuthenticationError: (Send-HPOVRequest:String) [New-HPOVExternalRepository], AuthSessionException
    + FullyQualifiedErrorId : InvalidOrTimedoutSession,New-HPOVExternalRepository
```

I don't expect an attempt to add a repository, even when it fails for some reason, to disconnect my session from the OneView appliance or Synergy Composer.

### Steps to reproduce

```PowerShell
PS C:\users\daveo> New-HPOVExternalRepository -Name "st052" -Hostname "gse-rose-st052.us.hpecorp.net" -Directory "/" -Http -Verbose
VERBOSE: [NEW-HPOVEXTERNALREPOSITORY] Bound PS Parameters: 
Key       Value                        
---       -----                        
Name      st052                        
Hostname  gse-rose-st052.us.hpecorp.net
Directory /                            
Http      True                         
Verbose   True                         



VERBOSE: [NEW-HPOVEXTERNALREPOSITORY] Called from: <ScriptBlock>
VERBOSE: [NEW-HPOVEXTERNALREPOSITORY] Verify auth
VERBOSE: [TEST-HPOVAUTH] Caller: New-HPOVExternalRepository
VERBOSE: [TEST-HPOVAUTH] Verify Auth for 15.113.228.235
VERBOSE: [TEST-HPOVAUTH] $Appliance is [HPOneView.Appliance.Connection]
VERBOSE: [TEST-HPOVAUTH] Received HPOneView.Appliance.Connection Object: 
ConnectionID Name           UserName      AuthLoginDomain Default
------------ ----           --------      --------------- -------
1            15.113.228.235 Administrator LOCAL           True   



VERBOSE: [NEWOBJECT] Bound PS Parameters: 
Key                Value
---                -----
ExternalRepository True 



VERBOSE: [NEWOBJECT] Called from: New-HPOVExternalRepository
VERBOSE: [NEW-HPOVEXTERNALREPOSITORY] Setting protocol to HTTP.
VERBOSE: [SEND-HPOVREQUEST] BEGIN
VERBOSE: [SEND-HPOVREQUEST] Bound PS Parameters: 
Key      Value                                                                                                                                                
---      -----                                                                                                                                                
uri      /rest/repositories                                                                                                                                   
method   POST                                                                                                                                                 
Hostname 15.113.228.235                                                                                                                                       
body     @{repositoryName=st052; userName=; password=; repositoryURI=http://gse-rose-st052.us.hpecorp.net//; repositoryType=FirmwareExternalRepo; base64Data=}



VERBOSE: [SEND-HPOVREQUEST] Process
VERBOSE: [SEND-HPOVREQUEST] Hostname value: 
ConnectionID Name           UserName      AuthLoginDomain Default
------------ ----           --------      --------------- -------
1            15.113.228.235 Administrator LOCAL           True   



VERBOSE: [SEND-HPOVREQUEST] Processing '15.113.228.235' appliance connection request. 1 of 1
VERBOSE: [SEND-HPOVREQUEST] Requested URI '/rest/repositories' to '15.113.228.235'
VERBOSE: [RESTCLIENT] Building new [System.Net.HttpWebRequest] object for POST https://15.113.228.235/rest/repositories
VERBOSE: [SEND-HPOVREQUEST] Body object found. Converting to JSON.
VERBOSE: [SEND-HPOVREQUEST] HTTP Method is POST. Removing 'ApplianceConnection' NoteProperty from object(s).
VERBOSE: InputObject is [PSCustomObject]. Copying...
VERBOSE: [SEND-HPOVREQUEST] Request Body: {"repositoryName":"st052","userName":"","password":"","repositoryURI":"http://gse-rose-st052.us.hpecorp.net//","repositoryType":"FirmwareExt
ernalRepo","base64Data":""}
VERBOSE: [SEND-HPOVREQUEST] Request: POST https://15.113.228.235/rest/repositories
VERBOSE: [SEND-HPOVREQUEST] Request Header 1: User-Agent = HPEOneview_PowerShell/4.00
VERBOSE: [SEND-HPOVREQUEST] Request Header 2: Content-Type = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 3: Accept = application/json
VERBOSE: [SEND-HPOVREQUEST] Request Header 4: X-API-Version = 800
VERBOSE: [SEND-HPOVREQUEST] Request Header 5: accept-language = en_US
VERBOSE: [SEND-HPOVREQUEST] Request Header 6: accept-encoding = gzip, deflate
VERBOSE: [SEND-HPOVREQUEST] Request Header 7: auth = [*****REDACTED******]
VERBOSE: [SEND-HPOVREQUEST] Request Header 8: Host = 15.113.228.235
VERBOSE: [SEND-HPOVREQUEST] Net.WebException Error caught
VERBOSE: [SEND-HPOVREQUEST] Exception Object: 

PSMessageDetails      : 
Exception             : System.Net.WebException: The remote server returned an error: (401) Unauthorized.
                           at System.Net.HttpWebRequest.GetResponse()
                           at CallSite.Target(Closure , CallSite , WebRequest )
TargetObject          : 
CategoryInfo          : NotSpecified: (:) [], WebException
FullyQualifiedErrorId : WebException
ErrorDetails          : 
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Send-HPOVRequest<Process>, C:\Program Files\WindowsPowerShell\Modules\HPOneView.410\4.10.1732.2495\HPOneView.410.psm1: line 4811
                        at New-HPOVExternalRepository<Process>, C:\Program Files\WindowsPowerShell\Modules\HPOneView.410\4.10.1732.2495\HPOneView.410.psm1: line 22321
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {}




VERBOSE: [SEND-HPOVREQUEST] Exception Message: The remote server returned an error: (401) Unauthorized.
VERBOSE: [SEND-HPOVREQUEST] InnerException FullyQualifiedErrorId: 
VERBOSE: [SEND-HPOVREQUEST] InnerException Message: 
VERBOSE: [SEND-HPOVREQUEST] Exception
VERBOSE: [SEND-HPOVREQUEST] Getting Error Response
VERBOSE: [SEND-HPOVREQUEST] ERROR RESPONSE: {"errorCode":"INVALID_REPOSITORY_CREDENTIALS","message":"Unable to access the content on the webserver due to invalid credentials or unava
ilability of access permissions.","details":"","recommendedActions":["Check the credentials and ensure that the webserver has the correct access permissions configured and try again.
"],"errorSource":null,"nestedErrors":[],"data":{}}
VERBOSE: [SEND-HPOVREQUEST] Response Status: HTTP 401 [Unauthorized]
VERBOSE: [SEND-HPOVREQUEST] Response Header: Keep-Alive = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Connection = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Transfer-Encoding = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Cache-Control = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Content-Type = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Date = 
VERBOSE: [SEND-HPOVREQUEST] Response Header: Server = 
VERBOSE: [SEND-HPOVREQUEST] HTTP 401 error caught.
VERBOSE: [NEW-ERRORRECORD] Building ErrorRecord object
VERBOSE: [SEND-HPOVREQUEST] Cleaning up HttpWebRequest
New-HPOVExternalRepository : [Send-HPOVRequest]: Your session has timed out or is not valid. Please use Connect-HPOVMgmt to authenticate to your appliance.
At line:1 char:1
+ New-HPOVExternalRepository -Name "st052" -Hostname "gse-rose-st052.us ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : AuthenticationError: (Send-HPOVRequest:String) [New-HPOVExternalRepository], AuthSessionException
    + FullyQualifiedErrorId : InvalidOrTimedoutSession,New-HPOVExternalRepository
 ```

I see the "Unauthorized" message in the verbose output, but I still don't think this failure should disconnect my session.

### Version Information
HPE OneView PowerShell Library Version (`Get-HPOVVersion` or `$PSLibraryVersion`):
HPE OneView Appliance Version (`Get-HPOVVersion -ApplianceVer`):  
Output from `$PSVersionTable` on your Windows Host:
```PowerShell
$PSLibraryVersion

> $PSLibraryVersion

15.113.228.235                      LibraryVersion Path                                                                   
--------------                      -------------- ----                                                                   
ApplianceVersion: 4.10.01.348545.00 4.10.1732.2495 C:\Program Files\WindowsPowerShell\Modules\HPOneView.410\4.10.1732.2495

> $PSVersionTable

Name                           Value                                                                                                                                                 
----                           -----                                                                                                                                                 
PSVersion                      5.1.17134.228                                                                                                                                         
PSEdition                      Desktop                                                                                                                                               
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                               
BuildVersion                   10.0.17134.228                                                                                                                                        
CLRVersion                     4.0.30319.42000                                                                                                                                       
WSManStackVersion              3.0                                                                                                                                                   
PSRemotingProtocolVersion      2.3                                                                                                                                                   
SerializationVersion           1.1.0.1                          
```

**Comment by ChrisLynchHPE** [comment]
This condition isn't handled correctly for sure.  I'll have it addressed in the next library update.

**Comment by ChrisLynchHPE** [comment]
This is fixed in [Release 4.10.1802.1882](https://github.com/HewlettPackard/POSH-HPOneView/releases/tag/v4.10.1802.1882) and in [PowerShell Gallery](https://www.powershellgallery.com/packages/HPOneView.410/4.10.1802.1882).

Please verify before this issue is closed.

**Comment by daveolker** [comment]
I confirmed the new library fixes the issue.  Thanks!
